packages/x-scheduler/src/material/timeline/content/view-header/WeeksHeader.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { DateTime } from 'luxon';
+import { useStore } from '@base-ui-components/utils/store/useStore';
+import { getAdapter } from '../../../../primitives/utils/adapter/getAdapter';
+import { isWeekend } from '../../../../primitives/utils/date-utils';
+import { useDayList } from '../../../../primitives/use-day-list';
+import { useWeekList } from '../../../../primitives/use-week-list';
+import { selectors } from '../../../../primitives/use-timeline';
+import { useTimelineStoreContext } from '../../../../primitives/utils/useTimelineStoreContext';
+import './Headers.css';
+
+const adapter = getAdapter();
+
+export function WeeksHeader() {
+  const getDayList = useDayList();
+  const getWeekList = useWeekList();
+  const store = useTimelineStoreContext();
+
+  const visibleDate = useStore(store, selectors.visibleDate);
+
+  const weeks = React.useMemo(() => {
+    const weeksFirstDays = getWeekList({
+      date: visibleDate,
+      amount: 8,
+    });
+
+    const tempWeeks: { date: DateTime }[][] = [];
+    for (let i = 0; i < weeksFirstDays.length; i += 1) {
+      const weekStart = weeksFirstDays[i];
+      const weekDays = getDayList({ date: weekStart, amount: 'week' });
+      const weekDaysWithEvents = weekDays.map((date) => ({
```

Comment:
You can keep the processed date and avoid `adapter.format` to generate the key before.
Btw, the name is weird, I see no event 👀